testutil/keeper/unit_test_helpers.go

```diff
@@ -22,169 +23,173 @@ import (
 	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
 	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
 	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v3/modules/core/23-commitment/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v3/modules/light-clients/07-tendermint/types"
+	providertypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
 )
 
-// Constructs a provider keeper and context object for unit tests, backed by an in-memory db.
-func GetProviderKeeperAndCtx(t testing.TB) (providerkeeper.Keeper, sdk.Context) {
+// Parameters needed to instantiate an in-memory keeper
+type InMemKeeperParams struct {
+	Cdc            *codec.ProtoCodec
+	StoreKey       *storetypes.KVStoreKey
+	ParamsSubspace *paramstypes.Subspace
+	Ctx            sdk.Context
+}
 
-	cdc, storeKey, paramsSubspace, ctx := SetupInMemKeeper(t)
+// NewInMemKeeperParams instantiates in-memory keeper params with default values
+func NewInMemKeeperParams(t testing.TB) InMemKeeperParams {
+	storeKey := sdk.NewKVStoreKey(types.StoreKey)
+	memStoreKey := storetypes.NewMemoryStoreKey(types.MemStoreKey)
+
+	db := tmdb.NewMemDB()
+	stateStore := store.NewCommitMultiStore(db)
+	stateStore.MountStoreWithDB(storeKey, sdk.StoreTypeIAVL, db)
+	stateStore.MountStoreWithDB(memStoreKey, sdk.StoreTypeMemory, nil)
+	require.NoError(t, stateStore.LoadLatestVersion())
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
 
-	k := providerkeeper.NewKeeper(
-		cdc,
+	paramsSubspace := paramstypes.NewSubspace(cdc,
+		codec.NewLegacyAmino(),
 		storeKey,
-		paramsSubspace,
-		&MockScopedKeeper{},
-		&MockChannelKeeper{},
-		&MockPortKeeper{},
-		&MockConnectionKeeper{},
-		&MockClientKeeper{},
-		&MockStakingKeeper{},
-		&MockSlashingKeeper{},
-		&MockAccountKeeper{},
-		"",
+		memStoreKey,
+		paramstypes.ModuleName,
 	)
-	return k, ctx
-}
+	ctx := sdk.NewContext(stateStore, tmproto.Header{}, false, log.NewNopLogger())
 
-// Constructs a consumer keeper and context object for unit tests, backed by an in-memory db.
-func GetConsumerKeeperAndCtx(t testing.TB) (consumerkeeper.Keeper, sdk.Context) {
+	return InMemKeeperParams{
+		Cdc:            cdc,
+		StoreKey:       storeKey,
+		ParamsSubspace: &paramsSubspace,
+		Ctx:            ctx,
+	}
+}
 
-	cdc, storeKey, paramsSubspace, ctx := SetupInMemKeeper(t)
+// A struct holding pointers to any mocked external keeper needed for provider/consumer keeper setup.
+type MockedKeepers struct {
+	*MockScopedKeeper
+	*MockChannelKeeper
+	*MockPortKeeper
+	*MockConnectionKeeper
+	*MockClientKeeper
+	*MockStakingKeeper
+	*MockSlashingKeeper
+	*MockAccountKeeper
+	*MockBankKeeper
+	*MockIBCTransferKeeper
+	*MockIBCCoreKeeper
+}
```

Comment:
Could these concrete pointer types be replaced with their respective keeper interfaces?

Reply:
Good point! Fine to leave like this then